e2e/tests/performance/imagery.perf.spec.js

```diff
@@ -0,0 +1,173 @@
+        await page.goto('/', { waitUntil: 'networkidle' });
+
+        // Click a:has-text("My Items")
+        await page.locator('a:has-text("My Items")').click({
+            button: 'right'
+        });
+
+        // Click text=Import from JSON
+        await page.locator('text=Import from JSON').click();
+
+        // Upload Performance Display Layout.json
+        await page.setInputFiles('#fileElem', filePath);
+
+        // Click text=OK
+        await page.locator('text=OK').click();
+
+        await expect(page.locator('a:has-text("Performance Display Layout Display Layout")')).toBeVisible();
+
+        //Create a Chrome Performance Timeline trace to store as a test artifact
+        console.log("\n==== Devtools: startTracing ====\n");
+        await browser.startTracing(page, {
+            path: `${testInfo.outputPath()}trace.json`,
+            screenshots: true
+        });
+    });
+    test.afterEach(async ({ page, browser}) => {
+        console.log("\n==== Devtools: stopTracing ====\n");
+        await browser.stopTracing();
+
+        /* Measurement Section
+        / The following section includes a block of performance measurements.
+        */
+        //Get time difference between viewlarge actionability and evaluate time
+        await page.evaluate(() => (window.performance.measure("machine-time-difference", "viewlarge.start", "button-view-large-click-test")));
+
+        //Get All Performance Marks
+        const getAllMarksJson = await page.evaluate(() => JSON.stringify(window.performance.getEntriesByType("mark")));
+        const getAllMarks = JSON.parse(getAllMarksJson);
+        console.log('window.performance.getEntriesByType("mark")', getAllMarks);
+
+        //Get All Performance Measures
+        const getAllMeasuresJson = await page.evaluate(() => JSON.stringify(window.performance.getEntriesByType("measure")));
+        const getAllMeasures = JSON.parse(getAllMeasuresJson);
+        console.log('window.performance.getEntriesByType("measure")', getAllMeasures);
+
+    });
+    /* The following test will navigate to a previously created Performance Display Layout and measure the
+    /  following metrics:
+    /  - ElementResourceTiming
+    /  - Interaction Timing
+    */
+    test('Embedded View Large for Imagery is performant in Fixed Time', async ({ page, browser }) => {
+        const client = await page.context().newCDPSession(page);
+        // Tell the DevTools session to record performance metrics
+        // https://chromedevtools.github.io/devtools-protocol/tot/Performance/#method-getMetrics
+        await client.send('Performance.enable');
+        // Go to baseURL
+        await page.goto('/');
+
+        // To to Search Available after Launch
+        await page.locator('input[type="search"]').click();
+        await page.evaluate(() => (window.performance.mark("search-available")));
+        // Fill Search input
+        await page.locator('input[type="search"]').fill('Performance Display Layout');
```

Comment:
I'd save the selector string here as a const and reuse that since we'll be updating search soon, this will most likely change and then you only need to update it in one spot.

Reply:
@jvigliotta in this particular case, this is a locator pattern that will need to remain in place until we have a crash course on accessibility